test-clients/src/main/java/com/hedera/services/bdd/suites/token/UniqueTokenManagementSpecs.java

```diff
@@ -152,79 +185,132 @@ private HapiApiSpec happyPathFiveMintOneMetadata() {
 						mintToken(TOKEN_NAME, List.of(ByteString.copyFromUtf8("memo")))
 
 				).then(
-						getTokenNftInfo(TOKEN_NAME, 1).hasSerialNum(1).hasMetadata(ByteString.copyFromUtf8("memo")),
-						getTokenNftInfo(TOKEN_NAME, 2).hasSerialNum(2).hasMetadata(ByteString.copyFromUtf8("memo")),
-						getTokenNftInfo(TOKEN_NAME, 3).hasSerialNum(3).hasMetadata(ByteString.copyFromUtf8("memo")),
-						getTokenNftInfo(TOKEN_NAME, 4).hasSerialNum(4).hasMetadata(ByteString.copyFromUtf8("memo")),
-						getTokenNftInfo(TOKEN_NAME, 5).hasSerialNum(5).hasMetadata(ByteString.copyFromUtf8("memo"))
+						getTokenNftInfo(TOKEN_NAME, 1)
+								.hasSerialNum(1)
+								.hasMetadata(ByteString.copyFromUtf8("memo"))
+								.hasAccountID(TOKEN_TREASURY)
+								.hasTokenID(TOKEN_NAME),
+
+						getTokenNftInfo(TOKEN_NAME, 2)
+								.hasSerialNum(2)
+								.hasMetadata(ByteString.copyFromUtf8("memo"))
+								.hasAccountID(TOKEN_TREASURY)
+								.hasTokenID(TOKEN_NAME),
+
+						getTokenNftInfo(TOKEN_NAME, 3)
+								.hasSerialNum(3)
+								.hasMetadata(ByteString.copyFromUtf8("memo"))
+								.hasAccountID(TOKEN_TREASURY)
+								.hasTokenID(TOKEN_NAME),
+
+						getTokenNftInfo(TOKEN_NAME, 4)
+								.hasSerialNum(4)
+								.hasMetadata(ByteString.copyFromUtf8("memo"))
+								.hasAccountID(TOKEN_TREASURY)
+								.hasTokenID(TOKEN_NAME),
+
+						getTokenNftInfo(TOKEN_NAME, 5)
+								.hasSerialNum(5)
+								.hasMetadata(ByteString.copyFromUtf8("memo"))
+								.hasAccountID(TOKEN_TREASURY)
+								.hasTokenID(TOKEN_NAME),
+
+						getAccountInfo(TOKEN_TREASURY)
+								.hasToken(relationshipWith(TOKEN_NAME))
 				);
 	}
 
-	private HapiApiSpec sadPathWithFrozenToken() {
-		return defaultHapiSpec("sadPathWithFrozenToken")
+	private HapiApiSpec failsWithFrozenToken() {
```

Comment:
I will add this use-case for discussion tonight. It seems strange to create a token, set it to freeze by default and not be able to mint new NFTs

Reply:
I will leave this as unresolved for now

Reply:
I reviewed how fungible tokens are behaving. We should be able to mint new NFTs to the Treasury if `freeze by default` is `true`. The treasury has exception when we are changing its "supply" through minting